app/code/community/Sailthru/Email/controllers/ContentController.php

```diff
@@ -35,27 +35,52 @@ protected function _processItems($productIds, $store)
         $appEmulation = Mage::getSingleton('core/app_emulation');
         $emulateData = $appEmulation->startEnvironmentEmulation($store);
         $count = count($productIds);
-        $savedProducts = 0;
         $startTime = microtime(true);
 
+        $syncedProducts = 0;
+
         /** @var Mage_Catalog_Model_Resource_Product_Collection $collection */
         $collection = Mage::getModel('catalog/product')
             ->getCollection();
 
         $collection
             ->addAttributeToSelect("*")
             ->addIdFilter($productIds)
+            ->joinField('qty',
+                'cataloginventory/stock_item',
+                'qty',
+                'product_id=entity_id',
+                '{{table}}.stock_id=1',
+                'left')
             ->setPageSize(75);
 
         $page = 1;
         do {
             $collection->setCurPage($page++)->load();
-            foreach ($collection as $product) {
-                /** @var Mage_Catalog_Model_Product $product */
+
+            /** @var Mage_Catalog_Model_Product $product */
+            foreach ($collection->getItems() as $product) {
                 try {
-                    $success = Mage::getModel('sailthruemail/client_content')->saveProduct($product);
-                    if ($success) {
-                        $savedProducts++;
+                    if ($product->getStatus() == Mage_Catalog_Model_Product_Status::STATUS_ENABLED) {
+                        $success = Mage::getModel('sailthruemail/client_content')->saveProduct($product);
+                        if ($success) {
+                            $syncedProducts++;
+                        }
+
+                    } elseif ($product->getStatus() == Mage_Catalog_Model_Product_Status::STATUS_DISABLED) {
+                        try {
+                            $success = Mage::getModel('sailthruemail/client_content')->deleteProduct($product);
```

Comment:
When we delete a product from the Content Library we lose pageviews. Have you considered the pros and cons of delete versus expiring content in an ecomm environment? Does STATUS_DISABLED apply to products that might be temporarily out of stock for example?

Reply:
Disabled wouldn't usually be the equivalent of out of stock, I believe. 

Either way, I believe if it's a separate issue as this code doesn't touch the "delete" code directly.